Create subs-to-queue service (#192)
ocodista/trampar-de-casa: apps/auto-email-sender/subs-to-queue/README.md

# Subs to queue

> The objective of this service is send the necessary database information of `email-pre-renderer` and `email-composer` for a queue `email-pre-renderer-subs` and `email-sender-subs`.

Reviewer: ```suggestion
The goal of this service is to send all the active Subscribers from the database to the `email-pre-renderer` and `email-composer` through RabbitMQ queues: `email-pre-renderer-subs` and `email-sender-subs`.
```

## Prerequisites

> Before proceeding, you should have the following installed on your system:

Reviewer: ```suggestion
Before starting, you'll need:
```

- [RabbitMQ](https://www.rabbitmq.com/documentation.html)
- [Supabase](https://supabase.com/docs)
- [Node 18.16.0](https://nodejs.org/en/blog/release/v18.16.0)

> Refer to the official documentation for [Node.js](https://nodejs.org/), [RabbitMQ](https://www.rabbitmq.com/documentation.html), and [Supabase](https://supabase.io/) for installation guides.

Reviewer: ```suggestion
Refer to the official documentation for [Node.js](https://nodejs.org/), [RabbitMQ](https://www.rabbitmq.com/documentation.html), and [Supabase](https://supabase.io/) for installation guides.

Or, if you're smart, use Docker.
```

## Flow Diagram
![image](https://github.com/ocodista/trampar-de-casa/assets/68869379/e18ac657-a3d2-4d08-b95f-ef5f912d174c)

## Sequence Diagram

```mermaid
sequenceDiagram

participant S as Service
participant RC as ReadChunk
participant STIEC as sendToSubsInfoEmailComposer
participant STIEPR as sendToSubsInfoEmailPreRenderer
participant DB as Supabase
participant R as RabbitMQ

S->>DB: Fetch batch of subscribers
DB-->>S: Return batch of subscribers

loop For Each Chunk
	S->>RC: Subscribers[]
	RC->>STIEC: Send message with data { skillsId,startedWorkingAt,id,email,isConfirmed }
	RC->>STIEPR: Send message with data { id, email }
end
```

## Functions

### subsToQueue

- Create connection with RabbitMQ and Supabase
- Execute `getAllConfirmedSubscribersPaginated`

> - Create a loop for read all chunks

Reviewer: ```suggestion
- Create a loop to read all chunks
```
